addon/plugins/variable-plugin/variables/address.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  ADRES,
+  EXT,
+} from '@lblod/ember-rdfa-editor-lblod-plugins/utils/constants';
+import {
+  createEmberNodeSpec,
+  createEmberNodeView,
+  EmberNodeConfig,
+} from '@lblod/ember-rdfa-editor/utils/ember-node';
+import { v4 as uuidv4 } from 'uuid';
+import { DOMOutputSpec, PNode } from '@lblod/ember-rdfa-editor';
+import {
+  isVariable,
+  parseLabel,
+  parseVariableInstance,
+  parseVariableType,
+} from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/variable-plugin/utils/attribute-parsers';
+import {
+  findChildWithRdfaAttribute,
+  hasRDFaAttribute,
+} from '@lblod/ember-rdfa-editor-lblod-plugins/utils/namespace';
+import { span } from '@lblod/ember-rdfa-editor-lblod-plugins/utils/dom-output-spec-helpers';
+import {
+  contentSpan,
+  instanceSpan,
+  mappingSpan,
+  typeSpan,
+} from '../utils/dom-constructors';
+export class Address {
+  declare street: string;
+  declare zipcode: string;
+  declare municipality: string;
+  housenumber?: string | null;
+  constructor(
+    args: Pick<Address, 'street' | 'housenumber' | 'zipcode' | 'municipality'>,
+  ) {
+    Object.assign(this, args);
+  }
+
+  get formatted() {
+    const firstPart = this.housenumber
+      ? `${this.street} ${this.housenumber}`
+      : this.street;
+    const secondPart = `${this.zipcode} ${this.municipality}`;
+    return `${firstPart}, ${secondPart}`;
+  }
+
+  sameAs(other?: Address | null) {
+    if (other) {
+      return (
+        this.street === other.street &&
+        this.housenumber === other.housenumber &&
+        this.zipcode === other.zipcode &&
+        this.municipality === other.municipality
+      );
+    } else {
+      return false;
+    }
+  }
+
+  get hasHouseNumber() {
+    return !!this.housenumber;
+  }
+}
+
+export class ResolvedAddress extends Address {
+  addressRegisterId: string;
+
+  constructor(
+    args: Pick<
+      ResolvedAddress,
+      | 'street'
+      | 'housenumber'
+      | 'zipcode'
+      | 'municipality'
+      | 'addressRegisterId'
+    >,
+  ) {
+    super(args);
+    this.addressRegisterId = args.addressRegisterId;
+  }
+
+  static resolve(address: Address, addressRegisterId: string) {
+    return new ResolvedAddress({
+      addressRegisterId,
+      ...address,
+    });
+  }
+}
+
+const constructAddressNode = (address?: Address | ResolvedAddress) => {
```

Comment:
Any chance we can structure this a bit cleaner? 
output looks like this:
![image](https://github.com/lblod/ember-rdfa-editor-lblod-plugins/assets/16419862/61729d11-86b0-4584-94e3-f26a1aae08e7)

Would be nice if we could wrap the constituent data in the spans, so we avoid excessive use of the `content` attribute

Reply:
Yeah that would be cleaner indeed, I'll update it.

Reply:
The data is now wrapped inside the spans, instead of using the content attribute.